app-commons/src/main/scala/org/bitcoins/commons/jsonmodels/dlc/DLCMessage.scala

```diff
@@ -42,64 +44,382 @@ object DLCMessage {
+      outcome.serialized
+        .zip(nonces)
+        .map {
+          case (bytes, nonce) =>
+            pubKey.computeSigPoint(CryptoUtil.sha256(bytes).bytes, nonce)
+        }
+        .reduce(_.add(_))
+    }
+  }
+
+  case class SingleNonceOracleInfo(
+      pubKey: SchnorrPublicKey,
+      rValue: SchnorrNonce)
+      extends OracleInfo {
+    override def nonces: Vector[SchnorrNonce] = Vector(rValue)
+
+    override def verifySigs(
+        outcome: DLCOutcomeType,
+        sigs: Vector[SchnorrDigitalSignature]): Boolean = {
+      outcome match {
+        case EnumOutcome(outcome) =>
+          sigs.length == 1 && pubKey.verify(CryptoUtil.sha256(outcome).bytes,
+                                            sigs.head)
+        case UnsignedNumericOutcome(_) => false
+      }
+    }
 
     override def bytes: ByteVector = pubKey.bytes ++ rValue.bytes
 
-    def toTLV: OracleInfoV0TLV = OracleInfoV0TLV(pubKey, rValue)
+    override def toTLV: OracleInfoV0TLV = OracleInfoV0TLV(pubKey, rValue)
   }
 
-  object OracleInfo extends Factory[OracleInfo] {
+  object SingleNonceOracleInfo extends Factory[SingleNonceOracleInfo] {
 
-    val dummy: OracleInfo = OracleInfo(ByteVector.fill(64)(1))
-
-    override def fromBytes(bytes: ByteVector): OracleInfo = {
+    override def fromBytes(bytes: ByteVector): SingleNonceOracleInfo = {
       require(bytes.size == 64, s"OracleInfo is only 64 bytes, got $bytes")
 
       val pubkey = SchnorrPublicKey(bytes.take(32))
       val rValue = SchnorrNonce(bytes.drop(32))
 
-      OracleInfo(pubkey, rValue)
+      SingleNonceOracleInfo(pubkey, rValue)
     }
   }
 
-  case class ContractInfo(outcomeValueMap: Map[Sha256Digest, Satoshis])
-      extends NetworkElement
-      with MapWrapper[Sha256Digest, Satoshis] {
-    override def wrapped: Map[Sha256Digest, Satoshis] = outcomeValueMap
+  case class MultiNonceOracleInfo(
+      pubKey: SchnorrPublicKey,
+      nonces: Vector[SchnorrNonce])
+      extends OracleInfo {
+
+    override def verifySigs(
+        outcome: DLCOutcomeType,
+        sigs: Vector[SchnorrDigitalSignature]): Boolean = {
+      outcome match {
+        case EnumOutcome(_) => false
+        case UnsignedNumericOutcome(digits) =>
+          digits
+            .zip(sigs.take(digits.length))
+            .foldLeft(digits.length <= sigs.length) {
+              case (result, (digit, sig)) =>
+                result && pubKey.verify(CryptoUtil.sha256(digit.toString).bytes,
+                                        sig)
+            }
+      }
+    }
+
+    override def bytes: ByteVector =
+      pubKey.bytes ++ nonces.foldLeft(ByteVector.empty)(_ ++ _.bytes)
+
+    override def toTLV: OracleInfoV1TLV = OracleInfoV1TLV(pubKey, nonces)
+  }
+
+  object MultiNonceOracleInfo extends Factory[MultiNonceOracleInfo] {
+
+    override def fromBytes(bytes: ByteVector): MultiNonceOracleInfo = {
+      require(bytes.length > 32 && bytes.length % 32 == 0,
```

Comment:
size should be >= 64

Reply:
This is covered by the second condition if I understand your concern correctly

Reply:
0 nonces means 32 byte info which is definitely caught here, 1 is allowed